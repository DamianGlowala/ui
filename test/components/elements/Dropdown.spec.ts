import { describe, it, expect } from 'vitest'
import { UDropdown } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'
import uiDropdown from '../../../src/runtime/ui.config/elements/dropdown'

describe('Dropdown', () => {
  it.each([
    ['basic case', {}],
    ['with items', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }], ui: uiDropdown } }],
    ['with open', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }], ui: uiDropdown } }],
    ['with close', { props: { open: false, items: [{ label: 'label1' }, { label: 'label2' }], ui: uiDropdown } }],
    ['with hover mode', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }], mode: 'hover', ui: uiDropdown } }],
    ['width disabled', { props: { disabled: true, items: [{ label: 'label1' }, { label: 'label2' }], ui: uiDropdown } }],
    ['width slot', { slots: { default: 'Open slot' }, props: { open: true, items: [{ label: 'label1' }, { label: 'label2', slot: 'label 2 slot' }], ui: uiDropdown } }],
    ['width item slot', { props: { open: true, items: [{ label: 'label1', slot: 'label 1 slot', icon: 'i-heroicons-academic-cap' }, { label: 'label2', slot: 'label 2 slot' }], ui: uiDropdown } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Dropdown.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UDropdown)
    expect(html).toMatchSnapshot()
  })
})
