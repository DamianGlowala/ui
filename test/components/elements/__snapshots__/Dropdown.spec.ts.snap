// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown > renders basic case correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div disabled="false" id="headlessui-menu-button-0" aria-haspopup="menu" aria-expanded="false" data-headlessui-state="" class="" role="button"><button> Open </button></div>
  <!--v-if-->
</div>"
`;

exports[`Dropdown > renders with close correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div disabled="false" id="headlessui-menu-button-7" aria-haspopup="menu" aria-expanded="false" data-headlessui-state="" class="" role="button"><button> Open </button></div>
  <!--v-if-->
</div>"
`;

exports[`Dropdown > renders with disabled correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="">
  <div disabled="true" id="headlessui-menu-button-11" aria-haspopup="menu" aria-expanded="false" data-headlessui-state="" class="" role="button"><button disabled=""> Open </button></div>
  <!--v-if-->
</div>"
`;
